File: Frontend/T-Shirt/src/services/orderService.ts
import { api } from "./api";
import type { Order, CreateOrderPayload, PageResponse, OrderStatus } from "../types/order";

export const orderService = {
  createOrder: async (payload: CreateOrderPayload): Promise<Order> => {
    const response = await api.post("/orders", payload);
    return response.data;
  },

  getUserOrders: async (page = 0, size = 10): Promise<PageResponse<Order>> => {
    const response = await api.get("/orders", { params: { page, size } });
    return response.data;
  },

  getOrderByNumber: async (orderNumber: string): Promise<Order> => {
    const response = await api.get(`/orders/${orderNumber}`);
    return response.data;
  },

  cancelOrder: async (orderNumber: string, reason?: string): Promise<Order> => {
    const response = await api.post(`/orders/${orderNumber}/cancel`, null, {
      params: { reason },
    });
    return response.data;
  },

  // Admin APIs
  getAdminOrders: async (status?: OrderStatus, page = 0, size = 10): Promise<PageResponse<Order>> => {
    const response = await api.get("/admin/orders", { params: { status, page, size } });
    return response.data;
  },

  getAdminOrderByNumber: async (orderNumber: string): Promise<Order> => {
    const response = await api.get(`/admin/orders/${orderNumber}`);
    return response.data;
  },

  updateOrderStatus: async (orderNumber: string, status: OrderStatus, comment?: string): Promise<Order> => {
    const response = await api.patch(`/admin/orders/${orderNumber}/status`, { status, comment });
    return response.data;
  },
};

File: Backend/tshirt-store-backend/src/main/java/com/tshirtstore/service/OrderService.java
package com.tshirtstore.service;

import com.tshirtstore.dto.order.CreateOrderRequest;
import com.tshirtstore.dto.order.OrderResponse;
import com.tshirtstore.dto.order.UpdateOrderStatusRequest;
import com.tshirtstore.entity.OrderStatus;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface OrderService {
    OrderResponse createOrder(CreateOrderRequest request);
    Page<OrderResponse> getUserOrders(Pageable pageable);
    OrderResponse getOrderByNumber(String orderNumber);
    OrderResponse cancelOrder(String orderNumber, String reason);

    // Admin methods
    Page<OrderResponse> getAllOrders(OrderStatus status, Pageable pageable);
    OrderResponse getAdminOrderByNumber(String orderNumber);
    OrderResponse updateOrderStatus(String orderNumber, UpdateOrderStatusRequest request);
}
